plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/IgniteQueryRunner.java

```diff
@@ -0,0 +1,128 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.airlift.log.Logging;
+import io.trino.Session;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.intellij.lang.annotations.Language;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.airlift.units.Duration.nanosSince;
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.QueryAssertions.assertUpdate;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.tpch.TpchTable.CUSTOMER;
+import static io.trino.tpch.TpchTable.NATION;
+import static io.trino.tpch.TpchTable.ORDERS;
+import static io.trino.tpch.TpchTable.REGION;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class IgniteQueryRunner
+{
+    private static final Logger log = Logger.get(IgniteQueryRunner.class);
+
+    private static final String IGNITE_SCHEMA = "public";
+
+    private IgniteQueryRunner() {}
+
+    public static DistributedQueryRunner createIgniteQueryRunner(
+            TestingIgniteServer server,
+            Map<String, String> extraProperties,
+            Map<String, String> connectorProperties)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(createSession())
+                    .setExtraProperties(extraProperties)
+                    .build();
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
+            connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
+            connectorProperties.putIfAbsent("allow-drop-table", "true");
+
+            queryRunner.installPlugin(new IgniteJdbcPlugin());
+            queryRunner.createCatalog("ignite", "ignite", connectorProperties);
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    public static void copyFromTpchTables(
```

Comment:
Maybe instead of calling this directly you can structure it similar to how other query runners are. Add an argument to createIgniteQueryRunner which is a list of TPCH tables and only create the ones which are requested. Take a look at PostgreSqlQueryRunner for example.

Reply:
Move copy table into `TestIgniteConnectorTest`

Reply:
I'd prefer if this was consistent with other QueryRunners. It's easier for other people to make sense of code since everything is consistent across connectors that way. Also copying those large CREATE TABLE statements to multiple test classes takes up a lot of space and multiple places to keep in sync.

You can look at https://github.com/trinodb/trino/blob/5b8e7e3f99e95e2ce8e14b8fb7f60d8e9fb1e39e/plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/PhoenixQueryRunner.java#L104 for an example. Model the IgniteQueryRunner similar to PhoenixQueryRunner.

Reply:
Even if Ignite doesn't support CTAS you can keep the `CREATE TABLE` queries inside the QueryRunner class.
I'm not talking about using CTAS.

I was thinking more about trying to ensure that the CREATE TABLE statements don't need to copied to every single test class. The QueryRunner could have those in a Map<String, String> (key = tpch table name, value = CREATE TABLE) and the QueryRunner's setup should call `copyTable` in a loop and pick correct queries from the map and execute them.